### PR TITLE
update banking-bench to simulate mint event 

### PR DESCRIPTION
**Pull request by taozhu-chicago**
by specifying certain percentage of conflicting transactions with higher priority fee

#### Problem
Need tool to simulate banking buffer during nft mint event, where prioritized transactions with overlapping accounts take large chunk of buffer. 

#### Summary of Changes
1. updated banking-bench tool so its testing transaction include compute-budget::set_compute_unit_price ix.
2. add `--simulate-mint` to turn on simulation mode, in which `mint-txs-percentage` transactions in batch will have conflicting account, and higher priority fee. 
 3. add `--mint-txs-percentage [0..99]` to fine tune the ratio of conflicting tx vs non-conflict transactions in each batch.

Fixes #

### CODE REVIEW

File: banking-bench/src/main.rs [hunk 1]
```diff
@@ -36,6 +36,8 @@ use {
     },
 };
 
+const TRANSFER_TRANSACTION_COST: u32 = 1470;
```
Comment (apfitzge) [inline banking-bench/src/main.rs:39]: Where did this come from?

File: banking-bench/src/main.rs [hunk 2]
```diff
@@ -121,6 +144,34 @@ fn make_accounts_txs(
         .collect()
 }
 
+// In siumulate mint, 99% transactions in a batch are mint transaction (eg., have conflicting account and higher
```
Comment (apfitzge) [inline banking-bench/src/main.rs:147]: Fix the comment - variable % now, not a set 99%.

File: banking-bench/src/main.rs [hunk 3]
```diff
@@ -121,6 +144,34 @@ fn make_accounts_txs(
...
+// priority) and 1% regualr transactions (eg., non-conflict and low priority)
+fn is_simulated_mint_transaction(
+    simulate_mint: bool,
+    index: usize,
+    packets_per_batch: usize,
+    mint_txs_percentage: usize,
+) -> bool {
+    !simulate_mint || (index % packets_per_batch <= packets_per_batch * mint_txs_percentage / 100)
```
Comment (apfitzge) [inline banking-bench/src/main.rs:155]: if `simulate_mint` is false this returns true.

```suggestion
    simulate_mint && (index % packets_per_batch <= packets_per_batch * mint_txs_percentage / 100)
```

File: banking-bench/src/main.rs [hunk 4]
```diff
@@ -219,6 +274,18 @@ fn main() {
...
+        .arg(
+            Arg::new("simulate_mint")
+                .long("simulate-mint")
+                .takes_value(false)
+                .help("Simulate mint transactions to have hjigher priority"),
+        )
+        .arg(
+            Arg::new("mint_txs_percentage")
+                .long("mint-txs-percentage")
+                .takes_value(true)
```
Comment (apfitzge) [inline banking-bench/src/main.rs:286]: add in `.requires("simulate_mint")` so this isn't used if `--simulate-mint` isn't present.

**Review by apfitzge — DISMISSED** [review]
Overall looks good, seems like there's a bug in the should_simulate though.

File: banking-bench/src/main.rs (hunk 3 above)
Comment (taozhu-chicago) [reply banking-bench/src/main.rs:155]: dang, not sure your test cmd, but could it be the cause of weird 0% test result?

File: banking-bench/src/main.rs (hunk 1 above)
Comment (taozhu-chicago) [reply banking-bench/src/main.rs:39]: transfer transaction cost is static, grabbed the value from log, but added comment to explain

File: banking-bench/src/main.rs (hunk 2 above)
Comment (taozhu-chicago) [reply banking-bench/src/main.rs:147]: ✅

**Pull request merged**